test/gtest/conv_hip_igemm_xdlops.cpp

```diff
@@ -0,0 +1,168 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <tuple>
+
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+#include <miopen/env.hpp>
+#include "../conv2d.hpp"
+#include "get_handle.hpp"
+
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_TEST_FLOAT_ARG)
+
+static bool IsTestRunWith(const char* float_arg)
+{
+    assert(float_arg != nullptr);
+    const char* const p_envVar = miopen::GetStringEnv(MIOPEN_TEST_FLOAT_ARG{});
+    return (p_envVar != nullptr && std::strcmp(p_envVar, float_arg) == 0);
+}
+
+void GetArgs(const std::string& param, std::vector<std::string>& tokens)
+{
+    std::stringstream ss(param);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class ConfigWithInt8 : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+    std::vector<std::string> params;
+    switch(prec)
+    {
+    case miopenInt8: params = ConfigWithInt8::GetParam(); break;
+    case miopenHalf:
+    case miopenBFloat16:
+    case miopenFloat:
+    case miopenInt8x4:
+    case miopenInt32:
+    case miopenDouble:
+        FAIL() << "miopenHalf, miopenBFloat16, miopenFloat, miopenInt8x4, miopenInt32, "
+                  "miopenDouble data "
+                  "type not supported by "
+                  "conv_hip_igemm_mlir_xdlops test";
```

Comment:
> [Recommendation] At the time of adding the ConvHipImplicitGemmFwdXdlops solver (in #1692), only the INT8 type was supported, and the initial test set was of INT8 type only. Then there was #1803 which adds support for HALF and FLOAT, but that PR is missing the support of these new datatypes in tests. That is why the test lacks HALF and FLOAT support.
> 
> There is another problem with this test: it allows all solvers to work and therefore there there is no guarantee that namely the `ConvHipImplicitGemmFwdXdlops` is being tested. If there is a problem with it (i.e. it is disabled), then any other solver (i.e. Gemm) can do the job and test would succeed.
> 
> @xinlipn @JehandadKhan What about resolving there problems at once? Or let's address these in a separate PR?

@atamazov , thanks for the input. Does it make sense to open another PR if we can quickly move forward with the current one?

Reply:
@xinlipn 
> 1) Please correct me if I was wrong, it seems ConvHipImplicitGemmFwdXdlops is a different test

This `test_conv_hip_igemm_xdlops` (which is being refactored in this PR) was introduced together with the `ConvHipImplicitGemmFwdXdlops` solver, in #1692 (which was created by @JehandadKhan). Therefore I suspect that this test is intended to test specifically `ConvHipImplicitGemmFwdXdlops`. If this is so, then I recommend adding support for other datatypes, and limiting it to the `ConvHipImplicitGemmFwdXdlops` only.

Please discuss with @JehandadKhan in order to clarify things and to decide what needs to be done. My comment is only recommendation.

> 2) in case it's the same test as here, this PR can be used as 1:1 map from the original cTest and any other changes can be put in a separate PR to avoid confusion.

No objections!